compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -11061,7 +11061,12 @@ private STNode parseTemplateContent() {
         STToken nextToken = peek();
         while (!isEndOfBacktickContent(nextToken.kind)) {
             STNode contentItem = parseTemplateItem();
-            items.add(contentItem);
+            if (contentItem.kind == SyntaxKind.SEMICOLON_TOKEN) {
+                items.add(STNodeFactory.createLiteralValueToken(SyntaxKind.TEMPLATE_STRING, "{",
+                        contentItem.leadingMinutiae(), contentItem.trailingMinutiae()));
```

Comment:
Well, this is not the correct fix. This is a special casing for this scenario only. 

We have parser modes used in the lexer to achieve different tokenizations on different contexts.
In the `TEMPLATE` parser mode `;` will be tokenized as  a `TEMPLATE_STRING` whereas in `DEFAULT` parser mode it is tokenized as a `SEMICOLN_TOKN`. The problem here seems to be `;` token is tokenized in the wrong mode in the lexer.

Need to check why the `;` is not tokenized under the `TEMPLATE` mode.